protos/core.proto

```diff
@@ -263,6 +263,56 @@ message ExecutedTask {
   string error = 3;
 }
 
+message Timing {
+  int64 start_time_millis = 1;
```

Comment:
You might think I'm crazy, but I'd actually use `double` here. Long's are a PITA, doubles are safe for millis until like the year 200,000+ or something crazy. It also makes them much more readable in JSON format and anyone who wants to consume CLI output

Reply:
Yeah, I fairly strongly disagree here.
- part of the reason we use protos here is because we want to be language agnostic; I don't think using a type to represent a timestamp which we wouldn't use in any other language but to make things nicer in JS is a good idea
- presentation-layer concerns should be dealt with at the presentation-layer, not at the API layer
- I have once before dealt with a system that used `double`s to represent timestamps, it was a total pain; this might be fine in JS but bring those fields into any other language (and remember we do; in Java) and weird stuff starts to happen with floating point precision, and that causes some really difficult to track down bugs

Reply:
I'm sold :)I guess if we want nicer JSON, we can just write Long's as numbers when we we know it is safe to do so!